demos/asp-net-core/ITOps.ViewModelComposition/IRouteInterceptor.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using System.Threading.Tasks;
+
+namespace ITOps.ViewModelComposition
+{
+    public interface IRouteInterceptor
+    {
+        bool Matches(RouteData routeData, string httpVerb);
+    }
+
+    public interface ISubscribeToCompositionEvents : IRouteInterceptor
```

Comment:
oh yes, I like it as well. I missed this one, these files were created a few months ago with Udi on call, live typing in VS. I'll split them.